clusterman/kubernetes/kubernetes_cluster_connector.py

```diff
@@ -94,20 +94,23 @@ def __init__(self, cluster: str, pool: Optional[str], init_crd: bool = False) ->
             )
             self._label_selectors.append(f"{node_label_selector}={self.pool}")
 
-    def reload_state(self) -> None:
+    def reload_state(self, load_pods_info: bool = True) -> None:
         logger.info("Reloading nodes")
 
         self.reload_client()
 
         # store the previous _nodes_by_ip for use in get_removed_nodes_before_last_reload()
         self._prev_nodes_by_ip = copy.deepcopy(self._nodes_by_ip)
         self._nodes_by_ip = self._get_nodes_by_ip()
-        logger.info("Reloading pods")
-        (self._pods_by_ip, self._unschedulable_pods, self._excluded_pods_by_ip,) = (
-            self._get_pods_info_with_label()
-            if self.pool_config.read_bool("use_labels_for_pods", default=False)
-            else self._get_pods_info()
-        )
+        if load_pods_info:
+            logger.info("Reloading pods")
```

Comment:
I think the log statement should come before the if else clause?

Reply:
A better way is to add a verbose log of what strategy we are using to get the pods `self._pods_by_ip, self._unschedulable_pods, self._excluded_pods_by_ip` 

```py
if load_pods_info:
    logger.info("Reloading pods via label filtering")
else:
   logger.info("Reloading pods without any label filteribg")
```

Reply:
But that's outside the scope of this change.

Reply:
Cool!